backend/src/users/users.service.ts

```diff
@@ -1,38 +1,43 @@
 import { Injectable, StreamableFile } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
-import { DeleteResult, UpdateResult } from 'typeorm';
 import { PageDto, PageOptionsDto } from '@dtos/pages';
-import { createReadStream } from 'fs';
 import { EntityDoesNotExistError } from 'src/errors/entityDoesNotExist';
 import { CreateUserDto, QueryUserDto, UpdateUserDto } from '@dtos/users';
 import { User } from './entities/user.entity';
-import UserRepository from './repository/user.repository';
 import * as fs from 'fs';
 import { ResponseFriendshipDto, FriendshipTypeEnum } from '@dtos/friendships';
-import { FriendshipRepository } from 'src/relationships/friendships/repositories/friendship.repository';
 import { ResponseBlockDto, BlockTypeEnum } from '@dtos/blocks';
-import { BlockRepository } from 'src/relationships/blocks/repositories/blocks.repository';
+import { Block } from 'src/relationships/entities/block.entity';
+import { Friendship } from 'src/relationships/entities/friendship.entity';
+import { paginate } from 'src/common/paginate';
+import { DeleteResult, Repository, UpdateResult } from 'typeorm';
 
 @Injectable()
 export class UsersService {
   constructor(
-    @InjectRepository(UserRepository)
-    private usersRepository: UserRepository,
-    @InjectRepository(FriendshipRepository)
-    private friendshipRepository: FriendshipRepository,
-    @InjectRepository(BlockRepository)
-    private blockRepository: BlockRepository,
+    @InjectRepository(User)
+    private usersRepository: Repository<User>,
+    @InjectRepository(Friendship)
+    private friendshipRepository: Repository<Friendship>,
+    @InjectRepository(Block)
+    private blockRepository: Repository<Block>,
   ) {}
 
   async findAll(
     query?: QueryUserDto,
     pageOptions: PageOptionsDto = new PageOptionsDto(),
   ): Promise<PageDto<User>> {
-    return this.usersRepository.findAllPaginated(query, pageOptions);
+    const orderOptions = { id: pageOptions.order };
+    return paginate<Repository<User>, User>(
+      this.usersRepository,
+      query,
+      orderOptions,
+      pageOptions,
+    );
   }
 
   findOne(id: number): Promise<User> {
-    return this.usersRepository.findOne(id);
+    return this.usersRepository.findOneByOrFail({ id: id });
```

Comment:
Some services use `findOneBy`. I think it should be uniform across services. But it could be the object of a future issue.

Reply:
I created issue #147 to discuss this. For now maybe it is better to change this to `findOneBy` because otherwise it is being updated for TypeORM **and** changing behavior. What do you think?